src/matching.js

```diff
@@ -194,8 +209,9 @@ export function importMatchPatternSet(exportedMatchPatternSet) {
  * Future performance improvements could include:
  *   * Replacing the path matching implementation to eliminate regular expressions entirely.
  *   * Replacing the match pattern index, such as by implementing a trie.
+ * @private
  */
-class MatchPatternSet {
+class _MatchPatternSet {
```

Comment:
Prefixing with `_` seems odd to me since this isn't done with the private functions elsewhere (since it's advisory anyway I tend to think something like `@private` is enough but I understand why people do this sort of thing in the absence of type checking).

Not exporting from the module seems like a pretty good indication that it's private to the module also?

Reply:
The only reason I added that prefix was so that we could separately define a public `MatchPatternSet` type in JSDoc. Part of the broader problem of how we can better document class-like objects and events. 😕